examples/oneapi/cpp/source/decision_forest/df_reg_dense_batch.cpp

```diff
@@ -0,0 +1,85 @@
+/*******************************************************************************
+ * Copyright 2020 Intel Corporation
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *******************************************************************************/
+
+#include "oneapi/dal/algo/decision_forest.hpp"
+#include "oneapi/dal/data/accessor.hpp"
+#include "oneapi/dal/data/table.hpp"
+
+#include "example_util/utils.hpp"
+
+using namespace oneapi;
+namespace df = oneapi::dal::decision_forest;
+
+int main(int argc, char const *argv[]) {
+  constexpr std::int64_t row_count_train = 6;
+  constexpr std::int64_t row_count_test = 3;
+  constexpr std::int64_t column_count = 2;
+
+  const float x_train[] = {
+      -2.f, -1.f, -1.f, -1.f, -1.f, -2.f, +1.f, +1.f, +1.f, +2.f, +2.f, +1.f,
+  };
+  const float y_train[] = {
+      0.f, 0.f, 0.f, 1.f, 1.f, 1.f,
+  };
+
+  const float x_test[] = {
+      -1.f, -1.f, 2.f, 2.f, 3.f, 2.f,
+  };
+  const float y_test[] = {
+      0.f,
+      1.f,
+      1.f,
+  };
+
+  const auto x_train_table =
+      dal::homogen_table{row_count_train, column_count, x_train};
+  const auto y_train_table = dal::homogen_table{row_count_train, 1, y_train};
+
+  const auto x_test_table =
+      dal::homogen_table{row_count_test, column_count, x_test};
+  const auto y_test_table = dal::homogen_table{row_count_test, 1, y_test};
+
+  const auto df_desc =
+      df::descriptor<float, df::task::regression, df::method::default_dense>{}
+          .set_tree_count(tree_count)
+          .set_features_per_node(features_per_node)
+          .set_min_observations_in_leaf_node(min_observations_in_leaf_node)
+          .set_variable_importance_mode(df::variable_importance_mode::mdi)
+          .set_train_results_to_compute(
+              (std::uint64_t)
```

Comment:
It looks very strange and unnatural. Can we do without castes when setting parameters?

Reply:
Solved via adding method which accepts train_result_to_compute val  and operator | 
std::uint64_t operator|(oneapi::dal::decision_forest::train_result_to_compute value_left,
                        oneapi::dal::decision_forest::train_result_to_compute value_right) {
    return static_cast<std::uint64_t>(value_left) | static_cast<std::uint64_t>(value_right);
}